services/main.py

```diff
@@ -0,0 +1,83 @@
+from xpotato.graph_extractor.extract import GraphExtractor
+from xpotato.graph_extractor.extract import FeatureEvaluator
+from xpotato.dataset.utils import default_pn_to_graph
+from tuw_nlp.graph.utils import (
+    GraphFormulaMatcher,
+)
+
+import json
+import os
+
+from fastapi import FastAPI
+from pydantic import BaseModel
+
+
+FEATURE_PATH = os.getenv("FEATURE_PATH", None)
+GRAPH_FORMAT = os.getenv("GRAPH_FORMAT", None)
+LANG = os.getenv("LANG", None)
+
+assert FEATURE_PATH, "FEATURE_PATH is not set"
+assert GRAPH_FORMAT, "GRAPH_FORMAT is not set"
+assert LANG, "LANG is not set"
+
+
+def match_texts(text):
+    texts = text.split("\n")
+
+    graphs = list(EXTRACTOR.parse_iterable(
+        [text for text in texts], GRAPH_FORMAT))
+
+    predicted = []
+
+    feature_values = []
+    for k in FEATURES:
+        for f in FEATURES[k]:
+            feature_values.append(f)
+    matcher = GraphFormulaMatcher(
```

Comment:
this doesn't really need to be created again for every function call, right?

Reply:
Thanks I fixed it!